metrics/sources/kubelet/summary_decoders.go

```diff
@@ -0,0 +1,238 @@
+	decodeUserDefinedMetrics(containerMetrics, container.UserDefinedMetrics)
+
+	return containerMetrics
+}
+
+func decodeUptime(metrics *MetricSet, startTime time.Time) {
+	if startTime.IsZero() {
+		return
+	}
+
+	uptime := uint64(time.Since(startTime).Nanoseconds() / time.Millisecond.Nanoseconds())
+	addIntMetric(metrics, &MetricUptime, &uptime, startTime)
+}
+
+func decodeCPUStats(metrics *MetricSet, cpu *stats.CPUStats, startTime time.Time) {
+	if cpu == nil {
+		return
+	}
+
+	addIntMetric(metrics, &MetricCpuUsage, cpu.UsageCoreNanoSeconds, startTime)
+
+	// Need to translate from nanocores to millicores.
+	if cpu.UsageNanoCores != nil {
+		const nanoPerMilli uint64 = 1e6
+		cores := *cpu.UsageNanoCores / nanoPerMilli
+		addIntMetric(metrics, &MetricCpuUsageRate, &cores, startTime)
+	}
+}
+
+func decodeMemoryStats(metrics *MetricSet, memory *stats.MemoryStats, startTime time.Time) {
+	if memory == nil {
+		return
+	}
+
+	addIntMetric(metrics, &MetricMemoruUsage, memory.UsageBytes, startTime)
+	addIntMetric(metrics, &MetricMemoryWorkingSet, memory.WorkingSetBytes, startTime)
+	addIntMetric(metrics, &MetricMemoryPageFaults, memory.PageFaults, startTime)
+	addIntMetric(metrics, &MetricMemoryMajorPageFaults, memory.MajorPageFaults, startTime)
+}
+
+func decodeNetworkStats(metrics *MetricSet, network *stats.NetworkStats, startTime time.Time) {
+	if network == nil {
+		return
+	}
+
+	addIntMetric(metrics, &MetricNetworkRx, network.RxBytes, startTime)
+	addIntMetric(metrics, &MetricNetworkRxErrors, network.RxErrors, startTime)
+	addIntMetric(metrics, &MetricNetworkTx, network.TxBytes, startTime)
+	addIntMetric(metrics, &MetricNetworkTxErrors, network.TxErrors, startTime)
+}
+
+func decodeFsStats(metrics *MetricSet, fsKey string, fs *stats.FsStats) {
+	if fs == nil {
+		return
+	}
+
+	fsLabels := map[string]string{LabelResourceID.Key: fsKey}
+	if fs.UsedBytes != nil {
+		usage := LabeledMetric{
+			Name:       MetricFilesystemUsage.Name,
+			Labels:     fsLabels,
+			ValueType:  ValueInt64,
+			MetricType: MetricFilesystemUsage.Type,
+			IntValue:   int64(*fs.UsedBytes),
+		}
+		metrics.LabeledMetrics = append(metrics.LabeledMetrics, usage)
+	}
+
+	if fs.CapacityBytes != nil {
+		limit := LabeledMetric{
+			Name:       MetricFilesystemLimit.Name,
+			Labels:     fsLabels,
+			ValueType:  ValueInt64,
+			MetricType: MetricFilesystemLimit.Type,
+			IntValue:   int64(*fs.CapacityBytes),
+		}
+		metrics.LabeledMetrics = append(metrics.LabeledMetrics, limit)
+	}
+}
+
+func decodeUserDefinedMetrics(metrics *MetricSet, udm []stats.UserDefinedMetric) {
+	for _, metric := range udm {
+		mv := MetricValue{}
+		switch metric.Type {
+		case stats.MetricGauge:
+			mv.MetricType = MetricGauge
+		case stats.MetricCumulative:
+			mv.MetricType = MetricCumulative
+		default:
+			glog.V(4).Infof("Skipping %s: unknown custom metric type: %v", metric.Name, metric.Type)
+			continue
+		}
+
+		mv.ValueType = ValueFloat
+		mv.FloatValue = float32(metric.Value)
```

Comment:
I was under impression that cAdvisor's custom metrics were either int64 or float.